test/service/test_response.py

```diff
@@ -2613,76 +2624,125 @@ def tearDownClass(cls):
+                'within': [6000, 100_000],
+                'results': [
+                    self.CellCounts(88_000, **{'mouth mucosa': None}),
+                ],
+            },
+            'effectiveCellCount': {
+                'within': [3000, 7000],
+                'results': [
+                    self.CellCounts(3589, **{'brain': 1}),
+                    self.CellCounts(None, **{'Brain': 6210}),
+                ],
+            },
+        }
+        for facet, filter_data in test_data.items():
+            with self.subTest(facet=facet):
                 params = {
                     'catalog': self.catalog,
-                    'sort': 'cellCount',
-                    'order': 'asc' if ascending else 'desc'
+                    'sort': facet,
+                    'order': 'asc',
+                    'filters': json.dumps({
+                        facet: {
+                            'within': [
+                                filter_data['within']
+                            ]
+                        }
+                    })
                 }
                 url = self.base_url.set(path='/index/projects', args=params)
                 response = requests.get(str(url))
                 response.raise_for_status()
-                response_json = response.json()
-                actual_results = self._count_total_cells(response_json)
-                expected = ascending_results if ascending else list(reversed(ascending_results))
-                self.assertEqual(actual_results, expected)
-
-    def test_filter_by_cell_count(self):
-        """
-        Verify filtering by 'cellCount' filters the documents based on the total
-        number of cells in each document, using the sum of total cells when a
-        document contains more than one cell suspension inner entity.
-        """
-        params = {
-            'catalog': self.catalog,
-            'filters': json.dumps({
-                'cellCount': {
-                    'within': [
-                        [
-                            6000,
-                            9000
-                        ]
-                    ]
-                }
-            })
-        }
-        url = self.base_url.set(path='/index/projects', args=params)
-        response = requests.get(str(url))
-        response.raise_for_status()
-        response_json = response.json()
-        actual_results = self._count_total_cells(response_json)
-        expected_results = [
-            (1, 6210),
-            (2, 7738)
-        ]
-        self.assertEqual(actual_results, expected_results)
+                response = response.json()
+                actual = self.CellCounts.from_response(response)
+                self.assertEqual(actual, filter_data['results'])
+
+    class CellCounts:
+
+        def __init__(self, project_cells: Optional[int], **kwargs):
+            self.project_cells = project_cells
+            self.__dict__.update(kwargs)
```

Comment:
I did not mean to imply that each keyword arg becomes an attribute. Since organs have spaces, we should probably ditch the keyword arg approach and just use a attrs class.